domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java

```diff
@@ -458,6 +466,127 @@ public static void deleteById(String table, String id,
     setTotalRecords.invoke(collection, totalRecords);
     return collection;
   }
+
+  private static void streamTrailer(HttpServerResponse response, ResultInfo resultInfo) {
+    response.end(String.format("],%n  \"resultInfo\": %s%n}", Json.encode(resultInfo)));
+  }
+
+  private static <T> void streamGetResult(PostgresClientStreamResult<T> result,
+    String element, HttpServerResponse response) {
+    response.setStatusCode(200);
+    response.setChunked(true);
+    response.putHeader(HttpHeaders.CONTENT_TYPE, "application/json");
+    response.write("{\n");
+    response.write(String.format("  \"totalRecords\": %d,%n", result.resultInto().getTotalRecords()));
+    response.write(String.format("  \"%s\": [%n", element));
+    final int[] cnt = { 0 };
+    result.exceptionHandler(res -> {
+      String message = res.getMessage();
+      List<Diagnostic> diag = new ArrayList<>();
+      diag.add(new Diagnostic().withCode("500").withMessage(message));
+      result.resultInto().setDiagnostics(diag);
+      streamTrailer(response, result.resultInto());
+    });
+    result.endHandler(res -> streamTrailer(response, result.resultInto()));
+    result.handler(res -> {
+      if (cnt[0]++ > 0) {
+        response.write(String.format(",%n"));
+      }
```

Comment:
The count number is not needed. We only need to know whether it is the first or any other record. Can you change to a `first` boolean variable?
The array is misleading because there is a single value only. Can you change this hack to an Atomic... variable?

    AtomicBoolean first = new AtomicBoolean(true);

...

      if (first.get()) {
        first.set(false);
      } else {
        response.write(String.format(",%n"));
      }

Reply:
OK with that, I assume you also considered the shoter first.getAndSet ..
I also see that if there's an exception a comma will be omitted.. which generates invalid JSON